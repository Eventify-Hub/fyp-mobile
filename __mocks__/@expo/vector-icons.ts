import React from 'react';

export const Ionicons = () => null;
export const FontAwesome = () => null;
export const FontAwesome5 = () => null; // âœ… ADD THIS
export const MaterialIcons = () => null;
export const MaterialCommunityIcons = () => null;

export const Font = {
  isLoaded: () => true,
  loadAsync: () => Promise.resolve(),
};
