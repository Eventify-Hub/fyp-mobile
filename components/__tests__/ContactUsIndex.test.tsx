import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ContactUsScreen from '../contactus/ContactUsIndex';

import { Ionicons } from '@expo/vector-icons';

// Mock router
jest.mock('expo-router', () => ({
    useRouter: () => ({ back: jest.fn() }),
}));

describe('ContactUsScreen', () => {
    it('renders without crashing', () => {
        const { getByText } = render(<ContactUsScreen />);
        expect(getByText('Contact Us')).toBeTruthy();
    });

    it('renders the phone numbers correctly', () => {
        const { getByText } = render(<ContactUsScreen />);
        expect(getByText('📞 +92 333 1283810')).toBeTruthy();
        expect(getByText('📞 +92 300 1234567')).toBeTruthy();
    });

    it('renders the office address correctly', () => {
        const { getByText } = render(<ContactUsScreen />);
        expect(getByText('🏢 Office #42, Software Tech Park')).toBeTruthy();
        expect(getByText('📍 Islamabad, Pakistan')).toBeTruthy();
    });

    it('renders the back button', () => {
        const { UNSAFE_getByType } = render(<ContactUsScreen />);
        const icon = UNSAFE_getByType(Ionicons);
        expect(icon).toBeTruthy();
    });

    it('matches snapshot', () => {
        const tree = render(<ContactUsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders section headings', () => {
        const { getByText } = render(<ContactUsScreen />);
        expect(getByText('Phone Numbers')).toBeTruthy();
        expect(getByText('Office Address')).toBeTruthy();
    });

    it('renders phone section with correct count', () => {
        const { getAllByText } = render(<ContactUsScreen />);
        expect(getAllByText(/📞/).length).toBeGreaterThan(1);
    });




    it('renders correctly with all text blocks', () => {
        const { getByText } = render(<ContactUsScreen />);
        expect(getByText('Contact Us')).toBeTruthy();
        expect(getByText('Phone Numbers')).toBeTruthy();
        expect(getByText('Office Address')).toBeTruthy();
    });

    it('renders Ionicons back icon with correct props', () => {
        const { UNSAFE_getByType } = render(<ContactUsScreen />);
        const icon = UNSAFE_getByType(Ionicons);
        expect(icon.props.name).toBe('arrow-back');
        expect(icon.props.size).toBe(24);
        expect(icon.props.color).toBe('#000');
    });
});
