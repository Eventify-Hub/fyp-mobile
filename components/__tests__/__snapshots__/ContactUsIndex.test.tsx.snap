// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUsScreen matches snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#F8E9F0",
      "flexGrow": 1,
      "padding": 20,
      "paddingTop": 100,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "justifyContent": "center",
          "left": 20,
          "opacity": 1,
          "padding": 8,
          "position": "absolute",
          "top": 20,
          "zIndex": 1,
        }
      }
    />
    <Text
      style={
        {
          "color": "#780C60",
          "fontSize": 26,
          "fontWeight": "bold",
          "marginBottom": 30,
          "textAlign": "center",
        }
      }
    >
      Contact Us
    </Text>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "elevation": 2,
          "marginBottom": 20,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Phone Numbers
      </Text>
      <Text
        style={
          {
            "color": "#444",
            "fontSize": 16,
            "marginBottom": 6,
          }
        }
      >
        📞 +92 333 1283810
      </Text>
      <Text
        style={
          {
            "color": "#444",
            "fontSize": 16,
            "marginBottom": 6,
          }
        }
      >
        📞 +92 300 1234567
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "borderRadius": 12,
          "elevation": 2,
          "marginBottom": 20,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Office Address
      </Text>
      <Text
        style={
          {
            "color": "#444",
            "fontSize": 16,
            "marginBottom": 6,
          }
        }
      >
        🏢 Office #42, Software Tech Park
      </Text>
      <Text
        style={
          {
            "color": "#444",
            "fontSize": 16,
            "marginBottom": 6,
          }
        }
      >
        📍 Islamabad, Pakistan
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
